WireDomain/Sources/WireDomain/Event Decryption/MLSMessageDecryptor.swift

```diff
@@ -0,0 +1,168 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireAPI
+import WireDataModel
+
+// sourcery: AutoMockable
+/// Decrypt MLS messages.
+protocol MLSMessageDecryptorProtocol {
+
+    /// Decrypt a MLS message.
+    ///
+    /// - Parameter eventData: A payload containing the encrypted message.
+    /// - Returns: The payload containing the decrypted message.
+
+    func decryptedEventData(
+        from eventData: ConversationMLSMessageAddEvent
+    ) async throws -> ConversationMLSMessageAddEvent
+
+}
+
+struct MLSMessageDecryptor: MLSMessageDecryptorProtocol {
+
+    let mlsDecryptionService: any MLSDecryptionServiceInterface
+    let mlsService: any MLSServiceInterface
+    let conversationLocalStore: any ConversationLocalStoreProtocol
+
+    func decryptedEventData(
+        from eventData: ConversationMLSMessageAddEvent
+    ) async throws -> ConversationMLSMessageAddEvent {
+        let conversationID = eventData.conversationID
+
+        guard let mlsConversation = await conversationLocalStore.fetchConversation(
+            id: conversationID.uuid,
+            domain: conversationID.domain
+        ) else {
+            WireLogger.mls.error(
+                "failed to add mls message: conversation not found in db"
+            )
```

Comment:
if we throw I guess we can remove logging the error

Reply:
done in https://github.com/wireapp/wire-ios/pull/2164/commits/b31805a437cf70da6bd6c7ebba537c3bc769a76e